lib/cylc/graph_parser.py

```diff
@@ -123,7 +123,8 @@ class GraphParser(object):
     REC_PARAMETERS = re.compile(r'<[\w,=\-+]+>')
 
     # Detect and extract suite state polling task info.
-    REC_SUITE_STATE = re.compile('(\w+)(<([\w\.\-/]+)::(\w+)(:\w+)?>)')
+    REC_SUITE_STATE = re.compile(
+        r'(\w+)(<([\w\.\-/]+)::(\w[\w\-+%@]*)(:\w+)?>)')
```

Comment:
You've fixed the regex for the remote task name, but not the one for the local polling task name.